src/test/kotlin/org/elm/ide/intentions/RecordConstructorToLiteralIntentionTest.kt

```diff
@@ -0,0 +1,213 @@
+package org.elm.ide.intentions
+
+class RecordConstructorToLiteralIntentionTest : ElmIntentionTestBase(RecordConstructorToLiteralIntention()) {
```

Comment:
Can you add a test for a multi-line record constructor call?

e.g.
```elm
type alias MyTypeAlias = { a : String, b : Int }
foo =
    MyTypeAlias
        "A"
        1
```

Unfortunately we don't have a good way of reformatting these multi-line expressions, so I'd be fine with just making the intention unavailable when the func call expr contains a newline. Or you can try to make it work. I started to work on an abstraction to make this easier (`IndentedTextBuilder`), but it is only useful in some cases. You can see an example in `MapToFoldIntention`.

Reply:
Been having a look at this... the current code will still allow the intention for your example above, but will generate a single-line record literal, i.e. `foo = { a = "A", b = 1 }`. This is OK, but it would probably be better as a multi-line record literal. I got that working, and my tests all pass, with two exceptions: if your code isn't quite correctly formatted, the `IndentedTextBuilder` can end up generating invalid PSI structures. For example this works:
```
foo =
    MyType{-caret-}Alias
        "A"
        1
```
This becomes:
```
foo =
    { a = "A"
    , b = 1
    }
```
However the following will result in an invalid PSI structure:
```
foo = MyType{-caret-}Alias
    "A"
    1
```
I'll investigate more when I get a chance, but if it looks too hairy, would you rather just make it generate a single-line record literal, or would you prefer the whole intention to be disabled on multi-line functions? (I think I'd tend towards allowing the intention, but making it generate a single-line record literal.)